pyvista/core/filters/data_set.py

```diff
@@ -2169,7 +2169,7 @@ def contour(  # type: ignore[misc]
 
         if isinstance(scalars, str):
             scalars_name = scalars
-        elif isinstance(scalars, (Sequence, np.ndarray)):
+        elif isinstance(scalars, (Sequence, np.ndarray)) and not isinstance(scalars, str):
```

Comment:
In general this is a good approach but in this case it seems redundant and would probably be considered as such if we had `redundant-expr` enabled since `isinstance(scalars, str)` will always evaluate false here (it's already checked above)
```suggestion
        elif isinstance(scalars, (Sequence, np.ndarray))
```

We can leave it though and fix this if/when mypy complains later after adding this option (either via #6225 or #6970